cpp/dolfinx/fem/FiniteElement.h

```diff
@@ -64,12 +63,11 @@ class FiniteElement
   /// @param[in] value_shape Value shape for blocked element, e.g. `{3}`
   /// for a vector in 3D or `{2, 2}` for a rank-2 tensor in 2D. Can only
   /// be set for blocked scalar `element`. For other elements and scalar
-  /// elements it should be `std::nullopt`.
+  /// elements it should be empty vector.
   /// @param[in] symmetric Is the element a symmetric tensor? Should ony
   /// set for 2nd-order tensor blocked elements.
   FiniteElement(const basix::FiniteElement<geometry_type>& element,
-                std::optional<std::vector<std::size_t>> value_shape
-                = std::nullopt,
+                std::vector<std::size_t> value_shape = {}, int block_size = 1,
```

Comment:
The whole point of extra changes is to simplify this nontrivial code.

1. My motivation for having block size as an extra argument is that the logic to compute it already lives in basix UFL wrapper, so it is better to avoid duplicating this nontrivial assumption, pass the one from basix instead. It does not have to have a default value.
2. I do not see the point of using `std::optional<std::vector<>>`. Optional is okay for when its value might not be provided. But empty std::vector is semantically correct and nicely maps in Python to an empty list or empty array. For lists/tuples on the Python side one does not use `None` to indicate an empty tuple/list. So here I strongly disagree.

I'll revert the changes to previous commit so the regression is fixed.